OpenRA.Game/Graphics/Viewport.cs

```diff
@@ -214,7 +214,14 @@ public void Scroll(float2 delta, bool ignoreBorders)
 		}
 
 		// Rectangle (in viewport coords) that contains things to be drawn
-		static readonly Rectangle ScreenClip = Rectangle.FromLTRB(0, 0, Game.Renderer.Resolution.Width, Game.Renderer.Resolution.Height);
+		static Rectangle ScreenClip
+		{
+			get
+			{
+				return Rectangle.FromLTRB(0, 0, Game.Renderer.Resolution.Width, Game.Renderer.Resolution.Height);
```

Comment:
This will create additional GC pressure by creating and discarding many more objects.  You can keep this cached, and then use the OnResolutionChange event to update its value.